lib/active_record/patches/sqlite3_adapter.rb

```diff
@@ -0,0 +1,46 @@
+module ActiveRecord
+  module Patches
+    module SQLite3Adapter
+      def initialize(...)
+        # version 3.37 is required for strict typing support and the newest json operators
+        raise Litestack::InsufficientSQLiteVersionError if ::SQLite3::SQLITE_VERSION_NUMBER < 3037000
+        
+        super
+        
+        configure_connection unless options.fetch(:noinit, false)
+        
+        ::SQLite3::Database.prepend Litedb
+        ::SQLite3::Statement.prepend Litedb::Statement
+      end
+      
+      private
+      
+      def configure_connection
+        # time to wait to obtain a write lock before raising an exception
+        # https://www.sqlite.org/pragma.html#pragma_busy_timeout
+        raw_connection.busy_handler { sleep 0.001 }
```

Comment:
By default, [Rails will override this value by a `busy_timeout`](https://github.com/rails/rails/blob/8c23eaa4eb328e4d007507cee0cc4a46a3e803f0/activerecord/lib/active_record/connection_adapters/sqlite3_adapter.rb#L695) if so configured in `database.yml` (`busy_timeout` overrides previously stablished `busy_handler's`

I'm not quite sure what should be the take here, maybe a comment, or perhaps not letting it get overridden at all?

Reply:
Good point. I'll need to experiment and think more on this. I also wanted to hear from @oldmoe on why he prefers `busy_handler` to `busy_timeout` here, since it only sleeps. I'm sure there is a good reason, I just don't know enough yet about the intricacies. 

But we do need to have a clear answer to how we mix with `database.yml` configuration in this setup.